### PR TITLE
Enhanced Environment Variable Management with Improved CLI Behavior

### PR DESCRIPTION
**Pull request by nickbeckenham**
Summary
This PR introduces significant improvements to the lpop CLI tool, enhancing environment variable parsing, CLI behavior, and user experience. The changes include better handling of empty values, improved command inference, and enhanced documentation.
Key Changes
🔧 CLI Behavior Improvements
Smart Command Inference: The main lpop command now intelligently handles different input types:
File paths are treated as output destinations for getting variables
Variable assignments (KEY=value) are parsed and stored
Empty input defaults to current repository context
Enhanced Error Handling: Better error messages and user feedback throughout the CLI
Improved Type Safety: Added proper TypeScript types for all CLI operations
�� Environment File Parser Enhancements
Structured Parsing: Complete rewrite of the environment file parser with better structure handling
Empty Value Detection: Added support for detecting and reporting ignored entries with empty values
Comment Preservation: Enhanced comment handling and structure preservation
Type Safety: Introduced discriminated union types (VariableEntry, CommentEntry, EmptyEntry) for better type safety
🎯 User Experience Improvements
Better Feedback: Added warnings for ignored empty values with count reporting
Clearer Output: Improved console messages and status indicators
Enhanced Documentation: Updated README with clearer examples and usage patterns
🔒 Security & Configuration
Updated .env.example: Added PORT=3000 example for common Node.js development scenarios
GitHub Workflows: Streamlined CI/CD pipeline configuration
Issue Templates: Removed redundant GitHub issue templates for cleaner repository structure
Technical Details
Files Modified
Core Logic: src/cli.ts, src/env-file-parser.ts
Documentation: README.md, .env.example
Configuration: .gitignore, package.json, various platform packages
Testing: Comprehensive test updates in tests/cli.test.ts, tests/env-file-parser.test.ts
Breaking Changes
None - all changes are backward compatible
Testing
✅ All existing tests updated and passing
✅ New test coverage for enhanced parser functionality
✅ CLI behavior tests updated for new smart inference
Benefits
Better Developer Experience: More intuitive CLI behavior with smart input detection
Improved Reliability: Better handling of edge cases and empty values
Enhanced Security: Clearer documentation on secure practices
Type Safety: Full TypeScript coverage with proper discriminated unions
Migration Notes
No migration required - existing functionality remains unchanged while gaining new capabilities.
Ready for Review ✅
Tests Passing ✅
Documentation Updated ✅

### CODE REVIEW

File: tests/cli.test.ts [hunk 1]
```diff
@@ -55,19 +56,24 @@ describe('LpopCLI', () => {
...
 
     cli = new LpopCLI();
   });
 
+  afterEach(() => {
+    // Clear all mocks after each test to prevent state leakage
+    vi.clearAllMocks();
+  });
+
   afterEach(() => {
```
Comment (tombeckenham) [inline tests/cli.test.ts:77]: This is listed twice? Not sure you can do that

**Review by tombeckenham — APPROVED** [review]
